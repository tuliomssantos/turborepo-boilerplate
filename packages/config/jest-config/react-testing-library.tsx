import React from 'react'

import { render } from '@testing-library/react'

const customRender = (ui: React.ReactNode, options = {}) =>
  render(<React.Fragment>{ui}</React.Fragment>, {
    wrapper: ({ children }) => children,
    ...options,
  })

export * from '@testing-library/react'
export { customRender as render }
export { default as userEvent } from '@testing-library/user-event'
