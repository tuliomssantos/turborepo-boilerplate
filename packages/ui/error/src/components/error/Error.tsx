export default function Error({ message }: { message: string }) {
  console.error(message)
  return <div>Error</div>
}
