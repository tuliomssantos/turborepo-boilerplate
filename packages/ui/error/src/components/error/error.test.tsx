import { render, screen } from 'jest-common/react-testing-library'

import Error from './Error'

describe('Error', () => {
  it('should render an error', () => {
    render(<Error message="Something went wrong" />)
    const error = screen.getByText('Error')
    expect(error).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { asFragment } = render(<Error message="Something went wrong" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
