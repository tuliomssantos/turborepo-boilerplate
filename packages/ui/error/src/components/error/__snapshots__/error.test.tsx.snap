// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match snapshot 1`] = `
<DocumentFragment>
  <div>
    Error
  </div>
</DocumentFragment>
`;
